response/page_path.go

```diff
@@ -2,13 +2,127 @@ package response
 
 import (
 	"context"
+	"net/url"
+	"regexp"
+	"strings"
 
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
+var visualisationsEndpointRegexp = regexp.MustCompile(`(^/visualisations/.+?)/`)
+var bulletinsOrArticlesPathRegexp = regexp.MustCompile(`(.+/(bulletin|article)s/.+?/[^/]+)`)
+var methodologiesOrQMIsOrAdhocsPathRegexp = regexp.MustCompile(`.+/(methodologies|qmis|adhocs)/([^/]+)`)
+var endsWithFileExtRegexp = regexp.MustCompile(`/[^/]+\.\w+$`)
+var timeSeriesPathRegexp = regexp.MustCompile(`^(.*?/timeseries(?:/[^/]+){0,2})`)
+var datasetsPathRegexp = regexp.MustCompile(`^(.*?/datasets(?:/[^/]+){0,2})`)
+
 func getPagePath(ctx context.Context, uri string) string {
 	log.Info(ctx, "Calculating page path for "+uri)
```

Comment:
does it make sense to convert the incoming string to lowercase? Not sure the matchers are case insensitive?

Reply:
No, that would be wrong. It's only the protocol and domain that are case-insensitive, but everything after that is case-sensitive. `/test` and `/TEST` are two different paths.